src/NServiceBus.AcceptanceTests/Sagas/When_handling_concurrent_messages.cs

```diff
@@ -35,7 +35,8 @@ public async Task Should_not_overwrite_each_other(bool useOutbox)
 
                     var timeout = DateTime.UtcNow.AddSeconds(15);
 
-                    b.When(c => DateTime.UtcNow > timeout, (session, ctx) => session.SendLocal(new FinishMsg { OrderId = "12345" }));
+                    b.When(c => DateTime.UtcNow > timeout || c.SagaData != null,
```

Comment:
why do we need this `when` condition at all? It seems the saga handler already sends a `FinishMsg ` and this test doesn't seem to test concurrency on saga-completion messages?

Reply:
As far as I understand this condition is here to make sure the test completes after 15 seconds in any case but assumes that the subsequent assertions would then fail to indicate a problem. That's why I left that part as is. Another option would be to just let it run into a run timeout.

```
  .Run(TimeSpan.FromSeconds(15));
```

Reply:
I think `.Run(TimeSpan.FromSeconds(15));` would be much easier to grasp instead as it states the purpose more clearly (and it's not as confusing). However, I'll leave it to you to decide whether you want to make that change too or stick with the current approach.

Reply:
I force pushed a change that completely removes the when condition and decided to just leave the default run settings instead to avoid running into issues where downstream for some reason are taking longer.